libsolidity/ast/Types.cpp

```diff
@@ -596,6 +596,12 @@ TypePointer FixedPointType::binaryOperatorResult(Token::Value _operator, TypePoi
 		return commonType;
 	if (Token::isBitOp(_operator) || Token::isBooleanOp(_operator))
 		return TypePointer();
+	if (_operator == Token::Exp)
+	{
+		if (auto intType = dynamic_pointer_cast<IntegerType const>(commonType))
```

Comment:
I actually have the impression that we should not base this on the common type alone. Perhaps @axic should also give his opinion. I guess removing exp from this PR altogether might be a good thing to do.